File: example/iOS/AppRevealExample/Services/ExampleFeatureFlags.swift
import Foundation

#if DEBUG
import AppReveal
#endif

class ExampleFeatureFlags {

    static let shared = ExampleFeatureFlags()

    private let flags: [String: Any] = [
        "new_checkout_flow": true,
        "dark_mode_v2": false,
        "catalog_grid_layout": true,
        "order_tracking_map": false,
        "push_promo_enabled": true,
        "max_cart_items": 50,
        "api_version": "v2",
        "ab_test_group": "B",
    ]

    private init() {}

    func isEnabled(_ flag: String) -> Bool {
        flags[flag] as? Bool ?? false
    }
}

#if DEBUG
extension ExampleFeatureFlags: FeatureFlagProviding {
    func allFlags() -> [String: AnyCodable] {
        flags.mapValues { AnyCodable($0) }
    }
}
#endif

File: example/Android/app/src/main/kotlin/com/appreveal/example/services/ExampleFeatureFlags.kt
package com.appreveal.example.services

import com.appreveal.state.FeatureFlagProviding

object ExampleFeatureFlags : FeatureFlagProviding {

    private val flags: Map<String, Any> = mapOf(
        "new_checkout_flow" to true,
        "dark_mode_v2" to false,
        "catalog_grid_layout" to true,
        "order_tracking_map" to false,
        "push_promo_enabled" to true,
        "max_cart_items" to 50,
        "api_version" to "v2",
        "ab_test_group" to "B"
    )

    fun isEnabled(flag: String): Boolean = flags[flag] as? Boolean ?: false

    override fun allFlags(): Map<String, Any?> = flags.toMap()
}
